cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,130 @@
+ * 	search.exclude [string] 					- files and/or directories to be excluded (regexp)
+ * 	search.baseDirectory [string]			- base directory
+ * level [int]                        - used to print the output in a hierarchical way
+ */
+func FindRecursive(path string, defaultPerm defaultPermissions, search searchSettings, verbose bool, level int) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode, we are printing the output in a hierarchical way
+	if verbose {
+		level++
+	}
+
+	for _, f := range files {
+		name := f.Name()
+
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), search.baseDirectory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		if m, _ := regexp.MatchString(search.exclude, relativePath); m {
+			if verbose {
+				fmt.Printf("\x1b[33m%sExcluding %s\n\x1b[0m", strings.Repeat(" ", level), name)
+			}
+		} else {
+			if search.hidden || !strings.HasPrefix(name, ".") {
+				mode := f.Mode()
+				currentOwner, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+				currentGroup, _ := user.LookupGroupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Gid))
+
+				currentPerm := currentPermissions{
+					permissions:     mode.Perm().String(),
+					owner:           currentOwner.Username,
+					group:           currentGroup.Name,
+					hasCorrectOwner: currentOwner.Username == defaultPerm.owner,
+					hasCorrectGroup: currentGroup.Name == defaultPerm.group,
+				}
+
+				if mode.IsRegular() {
+					currentPerm.defaultPermissions = defaultPerm.file
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "f", fullPath, currentPerm, defaultPerm, verbose)
+				} else if mode.IsDir() {
+					currentPerm.defaultPermissions = defaultPerm.dir
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "d", fullPath, currentPerm, defaultPerm, verbose)
+					FindRecursive(fullPath, defaultPerm, search, verbose, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions returns true if the permissions are correct (false in another case)
+/* currentPermissions [string] - permissions that the file or directory has (i.e. -rwxrwxr-x)
+ * defaultPermissions [string] - permissions that should have (i.e. rwxrwxr-x)
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
+	return strings.Contains(currentPermissions, defaultPermissions)
+}
+
+// printOutput Prints the data in different formats according to the situation
+/* level [string] 														- used to print the output in a hierarchical way
+ * kind [string] 															- "f" if it is file or "d" in case of a directory
+ * fullPath [string] 				  								- file/directory full path (if verbose only use the name)
+ * currentPerm [currentPermissions]						- data to print relate
```

Comment:
this is not the right place to document the fields of these structures: the structures themselves should have comments (as noted above). See standard library documentation: https://golang.org/pkg/net/http/#Cookie

Use `go doc` for a quick commandline tool or  `godoc -http`  for the same web rendering as the stdlib.
So you can see how this doc looks like.

Once stuff lands on github it will be indexed and rendered by https://godoc.org/, so it's a good idea to have good looking docs.